lib/stytch/m2m.rb

```diff
@@ -13,11 +13,162 @@ class M2M
+    # M2M Access tokens contain a standard set of claims as well as any custom claims generated from templates.
+    #
+    # == Parameters:
+    # client_id::
+    #   The ID of the client.
+    #   The type of this field is +String+.
+    # client_secret::
+    #   The secret of the client.
+    #   The type of this field is +String+.
+    # scopes::
+    #   An array scopes requested. If omitted, all scopes assigned to the client will be returned.
+    #   The type of this field is nilable list of +String+.
+    #
+    # == Returns:
+    # An object with the following fields:
+    # access_token::
+    #   The access token granted to the client. Access tokens are JWTs signed with the project's JWKs.
+    #   The type of this field is +String+.
+    # expires_in::
+    #   The lifetime in seconds of the access token.
+    #   For example, the value 3600 denotes that the access token will expire in one hour from the time the response was generated.
+    #   The type of this field is +Integer+.
+    # token_type::
+    #   The type of the returned access token. Today, this value will always be equal to "bearer"
+    #   The type of this field is +String+.
+    def token(client_id:, client_secret:, scopes: nil)
+      request = {
+        grant_type: 'client_credentials',
+        client_id: client_id,
+        client_secret: client_secret
+      }
+      request[:scope] = scopes.join(' ') unless scopes.nil?
+
+      JSON.parse(post_request("/v1/public/#{@project_id}/oauth2/token", request), {:symbolize_names => true})
+    end
+    # ENDMANUAL(M2M::token)
+
+    # MANUAL(M2M::authenticate_token)(SERVICE_METHOD)
+    # +authenticate_token+ validates a M2M JWT locally.
+    #
+    # == Parameters:
+    # access_token::
+    #   The access token granted to the client. Access tokens are JWTs signed with the project's JWKs.
+    #   The type of this field is +String+.
+    # required_scopes::
+    #   A list of scopes the token must have to be valid.
+    #   The type of this field is nilable list of +String+.
+    # max_token_age::
+    #   The maximum possible lifetime in seconds for the token to be valid.
+    #   The type of this field is nilable +Integer+.
+    # == Returns:
+    # +nil+ if the token could not be validated, or an object with the following fields:
+    # scopes::
+    #   An array of scopes granted to the token holder.
+    #   The type of this field is list of +String+.
+    # client_id::
+    #   The ID of the client that was issued the token
+    #   The type of this field is +String+.
+    # custom_claims::
+    #   A map of custom claims present in the token.
+    #   The type of this field is +object+.
+    def authenticate_token(access_token:, required_scopes: nil, max_token_age: nil)
+      begin
+        decoded_jwt = authenticate_token_local(access_token)
+      rescue StandardError
+        # Could not authenticate locally
+        return nil
```

Comment:
Should we be raising the underlying error? Or is returning `nil` on failure more ruby-standard?

Reply:
To be honest, I'm really not sure.